container/dangerzone.py

```diff
@@ -181,64 +175,81 @@ def document_to_pixels(self) -> None:
+        result = run_command(
+            ["pdfinfo", pdf_filename],
+            error_message="PDF file is corrupted",
+            timeout_message=f"Extracting metadata from PDF timed out after {DEFAULT_TIMEOUT} seconds",
+        )
+        output_lines = result.stdout.decode().split("\n")
+        num_pages = None
+        for line in output_lines:
+            if "Pages:" in line:
+                search = re.search(r"\d+", line)
+                if search is not None:
+                    num_pages = int(search.group(0))
+        if num_pages is None:
+            raise ValueError("Number of pages could not be extraced from PDF")
 
         self.percentage += 2
 
         # Convert to RGB pixel data
-        percentage_per_page = 45.0 / len(page_filenames)
-        for page in range(1, len(page_filenames) + 1):
-            pdf_filename = f"/tmp/page-{page}.pdf"
-            png_filename = f"/tmp/page-{page}.png"
-            rgb_filename = f"/tmp/page-{page}.rgb"
-            width_filename = f"/tmp/page-{page}.width"
-            height_filename = f"/tmp/page-{page}.height"
-            filename_base = f"/tmp/page-{page}"
-
-            self.update_progress(
-                f"Converting page {page}/{len(page_filenames)} to pixels"
-            )
-
-            # Convert to png
-            run_command(
-                ["pdftocairo", pdf_filename, "-png", "-singlefile", filename_base],
-                error_message="Conversion from PDF to PNG failed",
-                timeout_message=f"Error converting from PDF to PNG, pdftocairo timed out after {DEFAULT_TIMEOUT} seconds",
-            )
+        percentage_per_page = 45.0 / num_pages
+        page_base = "/tmp/page"
+        for page in range(1, num_pages + 1):
+            ppm_filename = f"{page_base}-{page}.ppm"
+            rgb_filename = f"{page_base}-{page}.rgb"
+            width_filename = f"{page_base}-{page}.width"
+            height_filename = f"{page_base}-{page}.height"
+            filename_base = f"{page_base}-{page}"
 
-            # Save the width and height
-            with Image.open(png_filename, "r") as im:
-                width, height = im.size
-            with open(width_filename, "w") as f:
-                f.write(str(width))
-            with open(height_filename, "w") as f:
-                f.write(str(height))
+            self.update_progress(f"Converting page {page}/{num_pages} to pixels")
 
-            # Convert to RGB pixels
+            # Convert to PPM, which is essentially an RGB format
             run_command(
                 [
-                    "gm",
-                    "convert",
-                    png_filename,
-                    "-depth",
-                    "8",
-                    f"rgb:{rgb_filename}",
+                    "pdftoppm",
+                    pdf_filename,
+                    "-f",
+                    str(page),
+                    "-l",
+                    str(page),
+                    "-singlefile",
```

Comment:
There is a way to make this faster. We can run `pdftoppm` and output the whole PDF into `.ppm` files, instead of calling `pdftoppm` for each page. But it has one downside, to inform the user of the progress, we have to parse the commands' stdout (with the `-status` flag). Each converted file will be a line.

Reply:
I have now implemented this. But I didn't use this approach as the stdout should be plaintext and not not a binary format.

Reply:
I saw the implementation, thanks. I have commented on a very small part, but other than that, the code looks ok.